src/main/java/com/patternpedia/api/rest/controller/CandidateController.java

```diff
@@ -0,0 +1,148 @@
+import com.patternpedia.api.rest.model.PatternModel;
+import com.patternpedia.api.service.CandidateService;
+import com.patternpedia.api.service.PatternLanguageService;
+import io.swagger.v3.core.util.Json;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+//@CrossOrigin(allowedHeaders = "*", origins = "*")
+@RequestMapping(value = "/candidate", produces = "application/hal+json")
+public class CandidateController {
+
+    Logger logger = LoggerFactory.getLogger(CandidateController.class);
+
+    private CandidateService candidateService;
+    private PatternLanguageService patternLanguageService;
+    private ObjectMapper objectMapper;
+
+    public CandidateController(
+            CandidateService candidateService,
+            PatternLanguageService patternLanguageService,
+            ObjectMapper objectMapper
+    ) {
+        this.candidateService = candidateService;
+        this.patternLanguageService = patternLanguageService;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * GET Methods
+     */
+    @GetMapping(value = "/getAll")
+    CollectionModel<EntityModel<CandidateModel>> all() {
+
+        List<EntityModel<CandidateModel>> candidates = this.candidateService.getAllCandidates()
+                .stream()
+                .map(candidate -> new EntityModel<>(CandidateModel.from(candidate)))
+//                        getPatternViewLinks(patternView)))
+                .collect(Collectors.toList());
+
+        return new CollectionModel<>(candidates);
+    }
+
+    @GetMapping(value = "/getById/{candidateId}")
+    @PreAuthorize(value = "#oauth2.hasScope('read')")
+    Candidate getCandidateById(@PathVariable UUID candidateId) {
+        return this.candidateService.getCandidateById(candidateId);
+    }
+
+    @GetMapping(value = "/getByUri/{candidateUri}")
+    Candidate getCandidateByUri(@PathVariable String candidateUri) {
+        return this.candidateService.getCandidateByURI(candidateUri);
+    }
+
+    /**
+     * CREATE Methods
+     */
+    @PostMapping(value = "/create")
+//    @PostMapping(value = "/create")
+//    @PreAuthorize(value = "#oauth2.hasScope('write')")
+//    @CrossOrigin(exposedHeaders = "Location")
+    @ResponseStatus(HttpStatus.CREATED)
+    Candidate newCandidate(@RequestBody CandidateModel candidate) {
+        return this.candidateService.createCandidate(candidate);
+    }
+
+    @PostMapping(value = "/createComment/{candidateId}&{userId}")
```

Comment:
stick to REST conventions: "/candidates/{candidateId/comments/{userId}" (userId can stay an URL param if you'd like to)